### PR TITLE
[WIP]Update topology.py

### PR DESCRIPTION
**Pull request by amritrajck**
Conversion to Collection v3

^ Make sure to prefix the Title with [WIP]/[WIPTEST]

Purpose or Intent
=================

__Fixing__ X because it is currently doing Y which is incorrect. It should be doing Z.

or

__Extending__ X because I need it to do Y so that I can update / add more test cases; PR #123 depends on it.

or

__Adding tests__ for feature X to cover cases Y and Z. They were implemented in product version 1.2.3.

or

__Updating tests__ for feature X because the behavior changed in product version 1.2.3.

--
These should serve mainly as a guideline. The more related information you provide, the better. :wink:

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 4f95e85749f3cff300e4395e092998c2b450909d:

`cfme/containers/topology.py`:
- :bangbang: Line 12:1: **E303** *too many blank lines (3)*
- :heavy_exclamation_mark: Line 21:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 22:1: **W293** *blank line contains whitespace*
- :heavy_exclamation_mark: Line 25:40: **W291** *trailing whitespace*
- :red_circle: Line 26:9: **F841** *local variable 'ENTITY' is assigned to but never used*
- :red_circle: Line 26:9: **N806** *variable in function should be lowercase*
- :heavy_exclamation_mark: Line 31:1: **W293** *blank line contains whitespace*

Hmmm
:scream: You might like to check the lint of your code :point_up: 


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4f9a4ffc5c7d6f9ae0ffa8266815793cdc2087a7:

`cfme/containers/topology.py`:
- :red_circle: Line 25:9: **F841** *local variable 'ENTITY' is assigned to but never used*
- :red_circle: Line 25:9: **N806** *variable in function should be lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/containers/topology.py [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
+    datastore = attr.ib(default=None)
 
     @classmethod
     def load_topology_page(cls):
         navigate_to(cls, 'All')
 
 
+@attr.s
+class TopologyCollection(BaseCollection):
+    def __init__(self, appliance=None):
```
Comment (digitronik) [inline cfme/containers/topology.py]: @amritrajck, Please fallow collection-v3 guidelines.

File: cfme/containers/topology.py [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
 
 
-class Topology(TopologyMixin, Navigatable):
+class Topology(TopologyMixin, BaseEntity):
 
-    def __init__(self, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
+    region = attr.ib(default=None)
+    provider = attr.ib(default=None)
+    datastore = attr.ib(default=None)
```
Comment (digitronik) [inline cfme/containers/topology.py]: Question:
why region, provider and datastore?

File: cfme/containers/topology.py (hunk 1 above)
Comment (amritrajck) [reply cfme/containers/topology.py]: Could you please explain in this context? Do you mean there shouldn't be an init  and should leave the class currently empty?

File: cfme/containers/topology.py (hunk 2 above)
Comment (amritrajck) [reply cfme/containers/topology.py]: Oh that's a mistake. I was following this other collection. Will change it.

File: cfme/containers/topology.py (hunk 1 above)
Comment (digitronik) [reply cfme/containers/topology.py]: @amritrajck fallow this [collection design](http://cfme-tests.readthedocs.io/en/master/guides/collections.html) for better understanding.
Actully, respective `__init__()`  method is hidden inside the `BaseCollection` and `BaseEntity` which you already imported and inherited; so no need of it.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
Replacing with #9578, not including the DefaultDashboard update though. Thanks @amritrajck

**Pull request closed without merging**